# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsWindowMemoryReporter.cpp
# Commit: 108d5fb4418e
# Full Hash: 108d5fb4418e78e4dca0a746863e5784c6d80371
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-23 09:52:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r d8dea951a032 -r 108d5fb4418e dom/base/nsWindowMemoryReporter.cpp
--- a/dom/base/nsWindowMemoryReporter.cpp	Wed Apr 22 15:48:20 2020 +0000
+++ b/dom/base/nsWindowMemoryReporter.cpp	Wed Apr 22 15:48:17 2020 +0000
@@ -68,7 +68,7 @@
   }
 
   // Measure this window's descendents.
-  for (const auto& frame : bc->GetChildren()) {
+  for (const auto& frame : bc->Children()) {
     if (auto* childWin = nsGlobalWindowOuter::Cast(frame->GetDOMWindow())) {
       MOZ_TRY(AddNonJSSizeOfWindowAndItsDescendents(childWin, aSizes));
     }